From 33bd4f80235bb1fc28d1d10f7dabe9c83881f21b Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 15:37:23 +0800
Subject: [PATCH 259/296] AOSCOS: staging: sb105x: replace put_tty_driver()
 with tty_driver_kref_put()

Per commit 9f90a4ddef4e ("tty: drop put_tty_driver"), "put_tty_driver() is
an alias for tty_driver_kref_put()."

Rename all instances of `put_tty_driver()' to `tty_driver_kref_put()'.

Fixes: "AOSCOS: Revert "staging: sb105x: delete the driver""
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/staging/sb105x/sb_pci_mp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/sb105x/sb_pci_mp.c b/drivers/staging/sb105x/sb_pci_mp.c
index ccf02cfeb2fc..c5e1c39e7a7a 100644
--- a/drivers/staging/sb105x/sb_pci_mp.c
+++ b/drivers/staging/sb105x/sb_pci_mp.c
@@ -1765,7 +1765,7 @@ for (i = 0; i < drv->nr; i++) {
 out:
 	if (retval < 0) {
 		printk("Register tty driver Fail!\n");
-		put_tty_driver(normal);
+		tty_driver_kref_put(normal);
 		kfree(drv->state);
 	}
 
@@ -1784,7 +1784,7 @@ void mp_unregister_driver(struct uart_driver *drv)
     }
 
     tty_unregister_driver(normal);
-    put_tty_driver(normal);
+    tty_driver_kref_put(normal);
     drv->tty_driver = NULL;
 
 
-- 
2.48.1

